functorch/functorch/experimental/cond.py

```diff
@@ -117,21 +123,16 @@ def cond_autograd(pred, true_fn, false_fn, *operands):
     return cond(pred, true_fn, false_fn, *operands)
 
 
-def python_fallback(op):
-    def inner(pred, true_fn, false_fn, operands):
-        mode = torch._C._get_torch_dispatch_mode()
-        assert (mode is not None), "Mode should always be enabled for python fallback key"
-        with suspend_mode(mode):
-            res = trace_cond(mode, op, pred, true_fn, false_fn, operands)
-        return res
-
-    return inner
+@cond.py_impl(ProxyTorchDispatchMode)
+def inner(pred, true_fn, false_fn, operands):
+    mode = torch._C._get_torch_dispatch_mode()
+    assert (mode is not None), "Mode should always be enabled for python fallback key"
+    with suspend_mode(mode):
+        res = trace_cond(mode, cond, pred, true_fn, false_fn, operands)
+    return res
 
 
-cond = PyOperator('cond')
-cond.impl(DispatchKey.CPU, cond_dense)
-cond.impl(DispatchKey.Python, python_fallback(cond))
-cond.impl(DispatchKey.PythonTLSSnapshot, fallthrough_fn)
-cond.impl(DispatchKey.AutogradCPU, cond_autograd)
-cond.impl(DispatchKey.ADInplaceOrView, fallthrough_fn)
-cond.impl(DispatchKey.BackendSelect, fallthrough_fn)
+# TODO(voz): Make this automatic for keys, this is very ugly atm
+cond.fallthrough(DispatchKey.PythonTLSSnapshot)
+cond.fallthrough(DispatchKey.ADInplaceOrView)
+cond.fallthrough(DispatchKey.BackendSelect)
```

Comment:
So, re fallthrough, I think the easiest fix will be to add C bindings to query the set of keys in c++ which are fallthrough, and then override it as necessary based on python registrations. cond.fallthrough is not symmetric with how fallthrough registrations work in C++, which is that fallthrough is a special "function" you can register.

Reply:
Agreed, this one is important enough to leave an issue behind, let me do that rq